transactionoutbox-core/src/main/java/com/gruelbox/transactionoutbox/TransactionOutboxImpl.java

```diff
@@ -149,6 +149,36 @@ public boolean flush(Executor executor) {
         .orElse(false);
   }
 
+  @Override
+  public boolean flushTopic(Executor executor, String topicName) {
+    if (!initialized.get()) {
+      throw new IllegalStateException("Not initialized");
+    }
+    Instant now = clockProvider.get().instant();
+    List<CompletableFuture<Boolean>> futures = new ArrayList<>();
+    futures.add(
+        CompletableFuture.runAsync(() -> expireIdempotencyProtection(now), executor)
+            .thenApply(it -> false));
+
+    futures.add(
```

Comment:
With the removal of `expireIdempotencyProtection` you don't need all the `CompletableFuture` stuff.

Just
```
return doFlush(
                  tx ->
                      uncheckedly(
                          () ->
                              persistor.selectNextInSelectedTopics(
                                  tx, topicNames flushBatchSize, now)));
```